src/engagement_db_to_analysis/regression_analysis/multiple_imputation_regression_analysis.py

```diff
@@ -0,0 +1,142 @@
+
+
+def run_multiple_imputation_regression_analysis(participants, consent_withdrawn_field, rqa_analysis_config,
+                                                demog_analysis_configs):
+    """
+    Runs multiple-imputation regression analysis on one RQA configuration against multiple demographics.
+
+    The regression is run separately on each of the normal themes in the RQA code scheme, each against the normal codes
+    in the provided demographics.
+
+    :param participants: Participants to analyse.
+    :type participants: iterable of core_data_modules.traced_data.TracedData
+    :param consent_withdrawn_field: Field in each participants object which records if consent is withdrawn.
+    :type consent_withdrawn_field: str
+    :param rqa_analysis_config: Configuration for the RQA dataset to run the regression on.
+    :type rqa_analysis_config: core_data_modules.analysis.AnalysisConfiguration
+    :param demog_analysis_configs: Configuration for the demographic datasets to run the regression on.
+                                   TODO: The actual demographics are currently a hard-coded subset of what is provided
+                                         here. Derive automatically or from configuration in future.
+    :type demog_analysis_configs: list of core_data_modules.analysis.AnalysisConfiguration
+    :return: Dictionary of theme -> regression results table, formatted as a string.
+    :rtype: dict of str -> str
+            TODO: Return the regression results table as an object that can be inspected and formatted rather than a str
+    """
+    # If this RQA has no normal themes, exit early, so we don't spend time creating and imputing datasets unnecessarily.
+    if len(_normal_codes(rqa_analysis_config.code_scheme.codes)) == 0:
+        return dict()
+
+    # Initialise R
+    base = importr("base")  # R standard library
+    arm = importr("arm")  # Library for 'Data Analysis Using Regression and Multilevel/Hierarchical Models'
+    mice = importr("mice")  # Library for 'Multivariate Imputation by Chained Equations'
+    env = robjects.globalenv
+
+    # TODO: Derive these variables automatically or from configuration rather than from a hard-coded string.
+    demographic_datasets = {"gender", "disability", "recently_displaced", "age_category"}
+
+    data_frame = convert_participants_to_regression_data_frame(
+        participants, consent_withdrawn_field, rqa_analysis_config,
+        [d for d in demog_analysis_configs if d.dataset_name in demographic_datasets]
+    )
+
+    # Generate 20 copies of the input dataset, where each copy has had the missing data filled in with a different set
+    # of plausible values.
+    # Reset R's random number generator seed to ensure we get reproducible results.
+    log.info(f"Running multiple imputation for dataset '{rqa_analysis_config.dataset_name}'...")
+    base.set_seed(123)
+    multiple_imputed_data_frame = mice.mice(data_frame, m=20, printFlag=False)
```

Comment:
What is the basis for the choice of 20 for the number of imputations?  I think the optimal number of imputations can vary depending on the specific dataset and research question. Can we include it in our configs?